### PR TITLE
fix(progress-circular): add theming, simplify demo

### LINKED ISSUE #7570: progress-circular: theming not working probably.
Hi,

I have checked the master and I noticed all the theming issues are solved. Only the  md-progress-circular is stilling a little out of order by using the default blue rgb color indicating the theme is not set for the  md-progress-circular. Also the adapted size in my theme size seems to be ignored.

Thanks

### PR DESCRIPTION
**Pull request by crisbeto**
* Applies `$mdTheming` to the `progressCircular` component.
* Simplifies the `progressCircular` demo.

Closes #7570.

### CODE REVIEW

File: src/components/progressCircular/js/progressCircularDirective.js [hunk 1]
```diff
@@ -47,13 +47,22 @@ angular
     '$$rAF',
     '$window',
     '$mdProgressCircular',
+    '$mdTheming',
```
Comment (DevVersion) [inline src/components/progressCircular/js/progressCircularDirective.js]: Just a question related to the progress circular. Why are we using Strict DI here?

File: src/components/progressCircular/demoBasicUsage/index.html [hunk 2]
```diff
@@ -27,11 +27,11 @@ <h4 style="margin-top:10px">Determinate</h4>
   </p>
 
   <div layout="row" layout-sm="column" layout-align="space-around">
```
Comment (ThomasBurleson) [inline src/components/progressCircular/demoBasicUsage/index.html]: Let's show how different parent themes are propagated to the indicators.

File: src/components/progressCircular/js/progressCircularDirective.js [hunk 3]
```diff
@@ -47,13 +47,22 @@ angular
...
 
-function MdProgressCircularDirective($$rAF, $window, $mdProgressCircular, $mdUtil, $interval, $log) {
+function MdProgressCircularDirective(
+  $$rAF,
+  $window,
+  $mdProgressCircular,
+  $mdTheming,
+  $mdUtil,
+  $interval,
+  $log
```
Comment (DevVersion) [inline src/components/progressCircular/js/progressCircularDirective.js]: What about this parameter line breaking? I've never seen that before. I suggest the following [code style](https://github.com/angular/material/blob/master/src/components/autocomplete/js/autocompleteController.js#L10)

Comment (crisbeto) [inline src/components/progressCircular/js/progressCircularDirective.js]: Thanks, I wasn't really sure about how to break them up.

File: src/components/progressCircular/js/progressCircularDirective.js (hunk 1 above)
Comment (crisbeto) [inline src/components/progressCircular/js/progressCircularDirective.js]: Good question. I suppose I just went with it out of habit, I'll switch to the regular DI.

Comment (DevVersion) [inline src/components/progressCircular/js/progressCircularDirective.js]: Awesome, thanks :+1:

**Comment by crisbeto** [comment]
It's updated to include the dark theme and the notes by @DevVersion, @ThomasBurleson.

**Comment by DevVersion** [comment]
@crisbeto Theming and DI looks good to me :+1:

**Pull request closed without merging**